tiledb/sm/query/readers/sparse_index_reader_base.h

```diff
@@ -296,11 +296,72 @@ class SparseIndexReaderBase : public ReaderBase {
    * it is really required to determine if a query is incomplete from the client
    * side of a cloud request.
    */
-  struct ReadState {
+  class ReadState {
+   public:
+    /* ********************************* */
+    /*     CONSTRUCTORS & DESTRUCTORS    */
+    /* ********************************* */
+
+    ReadState() = delete;
+
+    ReadState(size_t frag_idxs_len)
+        : frag_idx_(frag_idxs_len) {
+    }
+
```

Comment:
Add doxygen for constructors. Mention which ones are only used by deserialization.

Reply:
Done!